src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -298,34 +1010,73 @@ public ResultSet getCrossReference(
-                new Pair<>(FK_NAME, BsonTypeInfo.STRING_TYPE_NAME),
-                new Pair<>(PK_NAME, BsonTypeInfo.STRING_TYPE_NAME),
-                new Pair<>(DEFERRABILITY, BsonTypeInfo.INT_TYPE_NAME));
-
-        // All fields in this result set are nested under the bottom namespace.
-        MongoJsonSchema botSchema = MongoJsonSchema.createEmptyObjectSchema();
-        botSchema.properties.put(BOT_NAME, resultSchema);
+        MongoJsonSchema botSchema =
+                createBottomSchema(
+                        new Pair<>(PKTABLE_CAT, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(PKTABLE_SCHEM, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(PKTABLE_NAME, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(PKCOLUMN_NAME, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(FKTABLE_CAT, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(FKTABLE_SCHEM, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(FKTABLE_NAME, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(FKCOLUMN_NAME, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(KEY_SEQ, BsonTypeInfo.INT_TYPE_NAME),
+                        new Pair<>(UPDATE_RULE, BsonTypeInfo.INT_TYPE_NAME),
+                        new Pair<>(DELETE_RULE, BsonTypeInfo.INT_TYPE_NAME),
+                        new Pair<>(FK_NAME, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(PK_NAME, BsonTypeInfo.STRING_TYPE_NAME),
+                        new Pair<>(DEFERRABILITY, BsonTypeInfo.INT_TYPE_NAME));
 
         return new MongoSQLResultSet(null, BsonExplicitCursor.EMPTY_CURSOR, botSchema);
     }
 
+    // Helper for getting the rows for the getPrimaryKeys result set. Given a (dbName, tableName)
+    // and an index doc, it produces a list of BsonDocuments where each document corresponds to a
+    // row in the result set representing a column in the index. This method is intended for use
+    // with the getFirstUniqueIndexResultSet method.
+    private List<BsonDocument> toGetPrimaryKeysDocs(
+            Pair<String, String> namespace, Document indexInfo) {
+        Document keys = indexInfo.get(INDEX_KEY_KEY, Document.class);
+        String indexName = indexInfo.getString(INDEX_NAME_KEY);
+        AtomicInteger pos = new AtomicInteger();
+
+        return keys.keySet()
+                .stream()
+                .map(
+                        key ->
+                                createSortableBottomBson(
+                                        // Per JDBC spec, sort by COLUMN_NAME.
+                                        GET_PRIMARY_KEYS_SORT_SPECS,
+                                        new BsonElement(
+                                                TABLE_CAT, new BsonString(namespace.left())),
+                                        new BsonElement(TABLE_SCHEM, new BsonString("")),
```

Comment:
Since there isn't an actual schema, it seems that `null` would be the appropriate value here.

Reply:
I went with empty since that's what we've been doing for the MySQLDatabaseMetaData implementation. I'll switch these to null though because that is probably the more accurate result value.